Meta/Lagom/Tools/CodeGenerators/LibWeb/BindingsGenerator/IDLGenerators.cpp

```diff
@@ -2153,14 +2153,35 @@ JS::ThrowCompletionOr<JS::Object*> @constructor_class@::construct(FunctionObject
             generator.set(".constructor_arguments", arguments_builder.string_view());
 
             generator.append(R"~~~(
-    auto impl = TRY(throw_dom_exception_if_needed(vm, [&] { return @fully_qualified_name@::create_with_global_object(window, @.constructor_arguments@); }));
+    auto construct_impl = [&] <typename T>() {
+        if constexpr (requires(T) { T::construct_impl(declval<::JS::Realm&>(), @.constructor_arguments@ ); }) {
```

Comment:
Pre-existing code, but any idea why this is `.constructor_arguments` and not just `constructor_arguments`?

Reply:
perhaps we could ask the initial author ;) <https://github.com/SerenityOS/serenity/commit/8c7d1986b80857dc17c5cb4cb9d1ea44ecb39c50#diff-eaf1c991aa111b94b7f4f3f896617842850e5bfa38b32bd5229c0ba7e2e718bdR969>

seems it could use some cleanup later